src/main/java/com/factset/sdk/utils/authentication/ConfidentialClient.java

```diff
@@ -98,6 +121,52 @@ public ConfidentialClient(final Configuration config)
             new TokenRequestBuilder().uri(this.providerMetadata.getTokenEndpointURI());
     }
 
+    /**
+     * Creates a new ConfidentialClient. When setting up the OAuth 2.0 client, this constructor reaches out to
+     * FactSet's well-known URI to retrieve metadata about its authorization server. This information along with
+     * information about the OAuth 2.0 client is stored and used whenever a new access token is fetched.
+     *
+     * @param config Configuration object.
+     * @param requestOptions Object that can configure options like proxy and SSL settings
+     * @throws AuthServerMetadataContentException If Meta Issuer or Meta Token Endpoint is missing.
+     * @throws AuthServerMetadataException        If reading from URL is unsuccessful.
+     * @throws NullPointerException               Unchecked exception, if config is null.
+     */
+    public ConfidentialClient(final Configuration config, RequestOptions requestOptions)
```

Comment:
We have some duplication of code now. Please find a way to share code.
Ideally we'd have the code only in one function/ctor and all other's call that one with defaults.